File: EjemploVBasic/frmAcceso.vb
Public Class frmAcceso
    Function Validar_Contrasena()
        If txtUsuario.Text = "Supervisor" And txtContrasena.Text = "456" Then
            MsgBox("Bienvenido", , "MedicalCenter")
            Me.Hide()
            frmMenuPrincipal.Show()
            txtUsuario.BackColor = Color.White
            txtContrasena.BackColor = Color.White
        Else
            MsgBox("Usuarioyt contraseña incorrectos", , "MedicaCenter")
            txtUsuario.Text = ""
            txtContrasena.Text = ""
            txtUsuario.Focus()
            txtUsuario.BackColor = Color.Violet
            txtContrasena.BackColor = Color.Violet
            'comentarios
        End If
    End Function

    Private Sub btnAceptar_Click(sender As Object, e As EventArgs) Handles btnAceptar.Click
        Validar_Contrasena()

    End Sub

    Private Sub txtContrasena_TextChanged(sender As Object, e As EventArgs) Handles txtContrasena.TextChanged

    End Sub

    Private Sub txtContrasena_KeyDown(sender As Object, e As KeyEventArgs) Handles txtContrasena.KeyDown
        If e.KeyCode = Keys.Enter Then
            Validar_Contrasena()
        End If
    End Sub
End Class

File: Ejemplos12clParcial/frmAcceso.cs
using System;
using System.Collections.Generic;
using System.ComponentModel;
using System.Data;
using System.Drawing;
using System.Linq;
using System.Text;
using System.Threading.Tasks;
using System.Windows.Forms;

namespace Ejemplos12clParcial
{
    public partial class frmAcceso : Form
    {
        public frmAcceso()
        {
            InitializeComponent();
        }

        private void frmAcceso_Load(object sender, EventArgs e)
        {

        }

        private void btnAceptar_Click(object sender, EventArgs e)
        {
            if (txtUsuario.Text == "Supervisor" & txtContrasena.Text == "456")
            {
                MessageBox.Show("Bienvenido", "MedialCenter");
                // comentarios aqui explique
                // usams código de C# instrucción if
                this.Hide();
                frmMenuPrincipal frm = new frmMenuPrincipal();
                frm.Show();
                txtUsuario.BackColor = Color.White;
                txtContrasena.BackColor = Color.White;
            }
            else
            {
                MessageBox.Show("Usuario y contraseña incorrectos", "MedicalCenter");
                txtUsuario.Text = "";
                txtContrasena.Text = "";
                txtUsuario.Focus();
                txtUsuario.BackColor = Color.Pink;
                txtContrasena.BackColor = Color.Pink;
            }
        }
    }
}
